python/lsst/sims/featureScheduler/mockTelem/mock_observatory.py

```diff
@@ -0,0 +1,485 @@
+        observation['moonDec'] = sun_moon_info['moon_dec']
+        observation['moonDist'] = _angularSeparation(observation['RA'], observation['dec'],
+                                                     observation['moonRA'], observation['moonDec'])
+        observation['solarElong'] = _angularSeparation(observation['RA'], observation['dec'],
+                                                       observation['sunRA'], observation['sunDec'])
+        observation['moonPhase'] = sun_moon_info['moon_phase']
+
+        observation['ID'] = self.obsID_counter
+        self.obsID_counter += 1
+
+        return observation
+
+    def check_mjd(self, mjd, cloud_skip=20.):
+        """See if an mjd is ok to observe
+        Parameters
+        ----------
+        cloud_skip : float (20)
+            How much time to skip ahead if it's cloudy (minutes)
+
+
+        Returns
+        -------
+        bool
+
+        mdj : float
+            If True, the input mjd. If false, a good mjd to skip forward to.
+        """
+        passed = True
+        new_mjd = mjd + 0
+        delta_t = (mjd-self.mjd_start)*24.*3600.
+        clouds = self.cloud_model.get_cloud(delta_t+0.)
+        # Maybe set this to a while loop to make sure we don't land on another cloudy time?
+        # or just make this an entire recursive call?
+        if clouds > self.cloud_limit:
+            passed = False
+            new_mjd = mjd + cloud_skip/60./24.
+        alm_indx = np.searchsorted(self.almanac.sunsets['sunset'], mjd) - 1
+        # at the end of the night, advance to the next setting twilight
+        if mjd > self.almanac.sunsets['sun_n12_rising'][alm_indx]:
+            passed = False
+            new_mjd = self.almanac.sunsets['sun_n12_setting'][alm_indx+1]
+        if mjd < self.almanac.sunsets['sun_n12_setting'][alm_indx]:
+            passed = False
+            new_mjd = self.almanac.sunsets['sun_n12_setting'][alm_indx+1]
+        # We're in a down night, advance to next night
+        if self.almanac.sunsets['night'][alm_indx] in self.down_nights:
+            passed = False
+            new_mjd = self.almanac.sunsets['sun_n12_setting'][alm_indx+1]
+        # recursive call to make sure we skip far enough ahead
+        if not passed:
+            while not passed:
+                passed, new_mjd = self.check_mjd(new_mjd)
+            return False, new_mjd
+        else:
+            return True, mjd
+
+    def observe(self, observation):
+        """Try to make an observation
+
+        Returns
+        -------
+        status : bool
+            Result of if the observation worked
+        observation : observation object
+            None if there was no observation taken. Completed observation with meta data filled in.
+        new_night : bool
+            Have we started a new night.
+        """
+
+        start_night = self.night.copy()
+
+        # Make sure the kinematic model is set to the correct mjd
+        t = Time(self.mjd, format='mjd')
```

Comment:
So, here we get the mjd from the observatory class itself. That means, if we wanted to schedule an observation some time in the future, we won't be able to. I think it would be better if we could add the mjd_obs to the `observation` and have a logic such that `if observation['mjd'] == 0. ` we fall back to the observatory time, `elif observation['mjd'] >= self.mjd` we use that time instead and `else`  we would raise and exception.

Reply:
I like adding an `mjd_requested` to the observation--then it could slew to the position, and if we're still before `mjd_requested`, skip ahead to that time and then execute the observation. I'll put that in on the next branch.